util/build_configs/cray-internal/chapel.modulefile.lua.template

```diff
@@ -0,0 +1,174 @@
+--[[
+
+    file chapel module
+
+    (C) Copyright 2020 Hewlett Packard Enterprise Development LP
+
+]]--
+
+
+-- reasons to keep module from continuing --
+
+
+-- no need to conflict with chapel Lmod has a one name rule
+conflict("cray-mpich2", "xt-mpich2")
+prereq("craype")
+
+--[[ a hugepages module is required for XC, not required for shasta at present 09/30/2020
+prereq_any(
+            "craype-hugepages2M",
+            "craype-hugepages4M",
+            "craype-hugepages8M",
+            "craype-hugepages16M",
+            "craype-hugepages32M",
+            "craype-hugepages64M",
+            "craype-hugepages128M",
+            "craype-hugepages256M",
+            "craype-hugepages512M",
+            "craype-hugepages1G",
+            "craype-hugepages2G"
+)
+--]]
+
+
+-- local vars: define & assign --
+
+
+-- template variables ----------------------------------------------------------
+local MOD_LEVEL           = "@@{pkg_version}"       -- 1.23.0, e.g.
+local INSTALL_ROOT        = "@@{platform_prefix}"   -- former BASE_INSTALL_DIR
+--------------------------------------------------------------------------------
+
+-- set comm layer
+local CHPL_COMM = os.getenv("CHPL_COMM") or ""
+
+-- set host arch
+local CHPL_HOST_ARCH = os.getenv("CPU") or ""
+if CHPL_HOST_ARCH ~= "aarch64" and CHPL_HOST_ARCH ~= "x86_64" then
+    -- host arch is neither ARM-based CPU nor Cray-XC/HPE Cray EX
+    -- LmodError() will cause this module to exit
+    LmodError("Cannot determine vaild host CPU. CPU found was " .. tostring(CHPL_HOST_ARCH))
+end
+
+-- set host network platform
+local CHPL_HOST_PLATFORM = ""
+local this_module_path = myFileName()
+if string.find(this_module_path, "aries") then
```

Comment:
I wonder if the host_platform can/should be template-ized too?

Reply:
Yes.  But for now, "Lmod on XC".